### PR TITLE
Commit generated protobuf code, only regenerate on-demand

### LINKED ISSUE #41: Commit code generated by prost-build
Since v0.11 (4fea65bc82e33acd18e2f48c7f4e52384c06c995), `prost-build` requires `protoc` to be installed while compiling. This means any downstream application / library of `bitar` also needs it to compile.

Seeing how "the new recommendation for libraries is to commit their generated protobuf" (https://github.com/tokio-rs/prost/pull/657), it's probably best for bitar to ship the generated code, rather than generating it at compile-time.

### PR DESCRIPTION
**Pull request by MCOfficer**
closes #41 

Since v0.11 (https://github.com/oll3/bita/commit/4fea65bc82e33acd18e2f48c7f4e52384c06c995), `prost-build` requires `protoc` to be installed while compiling. This means any downstream application / library of bitar also needs it to compile.

This PR adds the generated code directly the repository, and only regenerates it on-demand.

Deciding when to regenerate the code is a bit of a pain - you can't use `cargo:rerun-if-changed`, since that will run directly after a clean check-out, and thus fail without `protoc`. The best compromise between elegance and practicality i found was to only regenerate when the generated file has been removed. I've added some comments to clarify this; since the file is rarely changed it shouldn't be too much of a hassle.

Alternative solutions to consider:

- use `rerun-if-changed` *and* check if `protoc` is in PATH. This "silent switch" could lead to build inconsistencies downstream, where the buildscript also runs.
- instead of shipping a `.rs` file, compile the protobuf files to a `.bin` using `protoc` and commit that. The buildscript would work as before (sourcing from the `.bin` instead of `.proto3`), without needing `protoc`. See for instance https://github.com/influxdata/pbjson/pull/79
- Ship `protoc` with this crate, despite the security concerns raised in https://github.com/tokio-rs/prost/pull/657. See [`protoc-prebuilt`](https://lib.rs/crates/protoc-prebuilt), [`protoc_bin_vendored`](https://lib.rs/crates/protoc-bin-vendored), [`protoc_fetcher`](https://lib.rs/crates/protoc-fetcher)

### CODE REVIEW

**Review by MCOfficer — COMMENTED** [review]
Should also remove `protoc` from the CI, but i want to hear your thoughts first.

**Review by oll3 — CHANGES_REQUESTED** [review]
Looks good! Added a single request/question.

Other than that I think the method you've selected seems like the best option given the alternatives you listed. I think it makes it simple enough to regenerate the file when wanted.

> Should also remove protoc from the CI, but i want to hear your thoughts first.
The only reason I can think of to keep it would be to validate the checked-in file, but don't know what such a test would be good for really. Anyway, it can be restored if ever needed so let's remove it.

File: bitar/build.rs [hunk 1]
```diff
@@ -1,3 +1,17 @@
 fn main() {
-    prost_build::compile_protos(&["proto/chunk_dictionary.proto"], &["proto/"]).unwrap();
+    println!("cargo:rerun-if-changed=proto/"); // Note: change this to cargo:: (double colon) once MSRV>=1.77
```
[the remark this answers is not in the document]
Comment (oll3) [reply bitar/build.rs:2]: > Anyways, I don't mind adding it or just dropping that line entirely. After all, it's only skipping the second check for whether the file exists.

True that, remove it then.

**Comment by oll3** [comment]
Thanks @MCOfficer !  :star:

**Pull request merged**